openedx/features/content_type_gating/tests/test_access.py

```diff
@@ -85,24 +93,23 @@ def _get_fragment_from_block(block, user_id, course, request_factory, mock_get_c
 
     # This method of fetching the block from the descriptor bypassess access checks
     problem_block = runtime.get_module(block)
-
     # Attempt to render the block, this should return different fragments if the content is gated or not.
     frag = runtime.render(problem_block, 'student_view')
     return frag
 
 
-def _assert_block_is_gated(block, is_gated, user_id, course, request_factory, has_upgrade_link=True):
+def _assert_block_is_gated(block, is_gated, user, course, request_factory, has_upgrade_link=True):
```

Comment:
Can you explain why you started requiring a user?

Reply:
the get_blocks call i added expects a user, so I pass user to this function and the get_blocks function and still pass user id to _get_fragment_from_block